### PR TITLE
feat(logger): adding compact log output format

### PR DESCRIPTION
**Pull request by dsavitskiy**
Default logger output format can be hard to follow because each line
has a verbose prefix of date, time and code location.
For development purposes, a alternative compact format is introduced.

### CODE REVIEW

**Review by paulyoong — APPROVED** [review]

File: io-engine/src/core/env.rs [hunk 1]
```diff
@@ -561,7 +565,7 @@ impl MayastorEnvironment {
     }
 
     /// initialize the logging subsystem
-    fn init_logger(&mut self) -> Result<()> {
+    fn init_logger(&mut self) {
```
Comment (tiagolobocastro) [inline io-engine/src/core/env.rs:568]: Any reason for not making failure an error now?

**Review by tiagolobocastro — APPROVED** [review]
Please sign your commits before merging.

File: io-engine/src/core/env.rs (hunk 1 above)
Comment (dsavitskiy) [reply io-engine/src/core/env.rs:568]: This result was never checked in a meaningful way: it was just unwraped, causing panic if spdk_log_set_flag() failed.

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors merge
